src/workerd/server/workerd-api.c++

```diff
@@ -241,7 +242,8 @@ struct NoopModuleRegistryObserver final : public jsg::ModuleRegistryObserver {
 
 kj::Own<jsg::ModuleRegistry> WorkerdApiIsolate::compileModules(
     jsg::Lock& lockParam, config::Worker::Reader conf,
-    Worker::ValidationErrorReporter& errorReporter) const {
+    Worker::ValidationErrorReporter& errorReporter,
+    config::Config::Reader serverConfig) const {
```

Comment:
I think it could make sense to pass `capnp::List<config::ExtensionBundle>::Reader extensions` instead of the whole config.